### ISSUE TITLE
documentation: should document limitations in 1.12 w/ USB ports and OSX

### ISSUE THREAD

**Issue by kenyee** [docker/for-mac#41]
### Information
When Docker 1.11 was run via docker-machine inside Virtualbox, we had the ability to share the USB port.  For Android developers, it was useful for setting up a build server and connecting Android devices to it so that you could run unit tests on the connected devices.

Now, it no longer appears to be possible w/ Docker 1.12.  I had to dig a lot before I could find out that this doesn't work at all w/ Docker 1.12 :(
https://forums.docker.com/t/usb-devices-with-private-beta-on-os-x/9951/4

Also, the Android emulators have compatibility issues when Docker 1.12 is running.
Genymotion doesn't work because Virtualbox can't run at the same time as OSX's Hypervisor.
The official Android emulators use QEMU which also currently isn't compatible w/ OSX's Hypervisor.

Basically, if you're an Android developer who is trying to run a dev version of the server or run a Jenkins build server on the same machine, it won't work.  You're better off running Genymotion emulators and the older Docker 1.11 on Virtualbox if you need to run everything on the same machine.

**Comment by avsm** [comment]
Thanks, this is indeed a regression with respect to the Docker Toolbox functionality.  The critical path for this is to determine what needs to happen for HAXM support so that the Android emulator can co-exist with xhyve, and then USB passthrough should be fairly straightforward.

**Comment by djs55** [comment]
Thanks for the report. I've created a PR to update the documentation.

Looking at the linked forum issue the pass-through arguments are of the form `--device=/dev/tty.usbserial-A903BD58:/dev/ttyUSB0` i.e. they're attempting pass-through of ttys rather than full USB devices.

Note that

- latest toolbox has Docker 1.12, so you don't have to downgrade your docker engine version
- it should be possible to run Virtualbox 5 at the same time as Docker for Mac -- there was a conflict between Virtualbox 4 and the OSX hypervisor framework but this has been fixed in Virtualbox 5.

I didn't realise the official Android emulators use QEMU -- this is worth investigating since any conflict with the OSX hypervisor framework should be fixable upstream.

The online docs describe how [Docker for Mac and Docker Toolbox can coexist](https://docs.docker.com/docker-for-mac/docker-toolbox/) -- this is the best we can do for now.
